edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/RegistryApiClient.cs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Http
+{
+    using System;
+    using System.Globalization;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class RegistryApiClient : IRegistryApiClient
+    {
+        internal const string MediaType = "application/json";
+        const string PutModuleOnBehalfOfUriTemplate = "devices/{0}/modules/$edgeHub/putModuleOnBehalfOf?api-version={1}";
+        const string GetModuleOnBehalfOfUriTemplate = "devices/{0}/modules/$edgeHub/getModuleOnBehalfOf?api-version={1}";
+        const string ListModulesOnBehalfOfUriTemplate = "devices/{0}/getModulesOnTargetDevice?api-version={1}";
+        const string DeleteModuleOnBehalfOfUriTemplate = "devices/{0}/deleteModuleOnBehalfOf?api-version={1}";
+        const string ApiVersion = "2020-06-30-preview";
+
+        readonly Option<IWebProxy> proxy;
+        readonly ITokenProvider tokenProvider;
+        readonly Uri baseUri;
+
+        public RegistryApiClient(string upstreamHostname, Option<IWebProxy> proxy, ITokenProvider tokenProvider)
+        {
+            this.UpstreamHostName = Preconditions.CheckNonWhiteSpace(upstreamHostname, nameof(upstreamHostname));
+            this.proxy = proxy;
+            this.tokenProvider = Preconditions.CheckNotNull(tokenProvider, nameof(tokenProvider));
+
+            this.baseUri = new UriBuilder(Uri.UriSchemeHttps, upstreamHostname).Uri;
+        }
+
+        internal string UpstreamHostName { get; }
+
+        public async Task<HttpResponseMessage> PutModuleOnBehalfOfAsync(string actorEdgeDeviceId, CreateOrUpdateModuleOnBehalfOfData requestData)
+        {
+            var requesteUri = new Uri(this.baseUri, string.Format(CultureInfo.InvariantCulture, PutModuleOnBehalfOfUriTemplate, actorEdgeDeviceId, ApiVersion));
```

Comment:
Do we need some kind of retry logic here, if the call fails because of a transient error?

Reply:
Keep it simple for now as RegistryProxyMiddleware don't have any retry logic.